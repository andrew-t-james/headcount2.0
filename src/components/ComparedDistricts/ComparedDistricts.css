.comparison-container {
  align-items: center;
  border: 1px solid orchid;
  display: flex;
  height: calc(100vh - 7rem);
  justify-content: space-between;
}

@media screen and (max-width: 1200px) {
  .comparison-container {
    flex-direction: column;
    justify-content: space-around;
  }
}