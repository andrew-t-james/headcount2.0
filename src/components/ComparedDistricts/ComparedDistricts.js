import React from 'react';
import PropTypes from 'prop-types';
import GraphCard from '../GraphCard/GraphCard';

import './ComparedDistricts.css';

const ComparedDistricts = ({ comparedDistricts, comparisonData }) => {
  const displaySelectedCards = comparedDistricts.map(district =>
    <GraphCard
      {...district}
      key={district.id}
    />
  );

  return (
    <section >
      <div className="comparison-container">
        {displaySelectedCards[0]}
        <h2>{comparisonData.compared}</h2>
        {displaySelectedCards[1]}
      </div>
    </section>
  );
};

ComparedDistricts.propTypes = {
  comparedDistricts: PropTypes.arrayOf(PropTypes.object).isRequired,
  comparisonData: PropTypes.object
};

export default ComparedDistricts;