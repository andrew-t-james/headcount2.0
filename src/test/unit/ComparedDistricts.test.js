import React from 'react';
import { shallow } from 'enzyme';
import ComparedDistricts from '../../components/ComparedDistricts/ComparedDistricts';

describe('ComparedDistrict unit test suite', () => {
  let wrapper;
  const mockDistricts = [
    {
      location: 'Adams',
      stats: {
        2004: 0.393
      }
    },
    {
      location: 'Adams',
      stats: {
        2004: 0.393
      }
    }
  ];

  const mockComparisonData = {};

  beforeEach(() => wrapper = shallow(
    <ComparedDistricts
      comparedDistricts={mockDistricts}
      comparisonData={mockComparisonData}
    />
  ));

  afterEach(() => wrapper.unmount());

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot when no props passed', () => {
    wrapper = shallow(
      <ComparedDistricts
        comparedDistricts={[]}
        comparisonData={{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
