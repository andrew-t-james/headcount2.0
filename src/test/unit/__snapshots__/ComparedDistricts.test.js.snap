// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparedDistrict unit test suite should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparedDistricts
    comparedDistricts={
        Array [
            Object {
              "location": "Adams",
              "stats": Object {
                "2004": 0.393,
              },
            },
            Object {
              "location": "Adams",
              "stats": Object {
                "2004": 0.393,
              },
            },
          ]
    }
    comparisonData={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="comparison-container"
>
        <GraphCard
                location="Adams"
                stats={
                        Object {
                                "2004": 0.393,
                              }
                }
        />
        <h2 />
        <GraphCard
                location="Adams"
                stats={
                        Object {
                                "2004": 0.393,
                              }
                }
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <GraphCard
            location="Adams"
            stats={
                        Object {
                                    "2004": 0.393,
                                  }
            }
/>,
          <h2 />,
          <GraphCard
            location="Adams"
            stats={
                        Object {
                                    "2004": 0.393,
                                  }
            }
/>,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "location": "Adams",
            "stats": Object {
              "2004": 0.393,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "location": "Adams",
            "stats": Object {
              "2004": 0.393,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="comparison-container"
>
          <GraphCard
                    location="Adams"
                    stats={
                              Object {
                                        "2004": 0.393,
                                      }
                    }
          />
          <h2 />
          <GraphCard
                    location="Adams"
                    stats={
                              Object {
                                        "2004": 0.393,
                                      }
                    }
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <GraphCard
              location="Adams"
              stats={
                            Object {
                                          "2004": 0.393,
                                        }
              }
/>,
            <h2 />,
            <GraphCard
              location="Adams"
              stats={
                            Object {
                                          "2004": 0.393,
                                        }
              }
/>,
          ],
          "className": "comparison-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "location": "Adams",
              "stats": Object {
                "2004": 0.393,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "location": "Adams",
              "stats": Object {
                "2004": 0.393,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`ComparedDistrict unit test suite should match snapshot when no props passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparedDistricts
    comparedDistricts={Array []}
    comparisonData={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="comparison-container"
>
        <h2 />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <h2 />,
          undefined,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        undefined,
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="comparison-container"
>
          <h2 />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <h2 />,
            undefined,
          ],
          "className": "comparison-container",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          undefined,
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
